### PR TITLE
Create client component store with DB connection + fix and integrate auth guards

### PR DESCRIPTION
**Pull request by m-a-s-h-e-d**
### Description

This branch creates a component store for state management on the client page (which can be copied and used for all the other respective pages) as well as creates queries for Supabase DB connection.

This branch additionally fixes the issue with authentication and adds guards to routes to ensure a user is logged in. It checks if a session flag is enabled in the SessionStorage which is created and removed on authentication status updates. It is not possible to spoof another account with this flag due to Supabase + PostgreSQL integrated RLS which requires the authenticated user to make a request in order to access its' resources.

### CODE REVIEW

File: dna-frontend/src/app/pages/businesses/businesses.store.ts
Comment (m-a-s-h-e-d) [inline dna-frontend/src/app/pages/businesses/businesses.store.ts:1]: This needs to be replaced when working on the Businesses page, please reference `client.store.ts`

**Review by nng32 — CHANGES_REQUESTED** [review]
The client link on the sidebar no longer works because the client route changed

**Review by nng32 — APPROVED** [review]
lgtm!!!!

**Comment by m-a-s-h-e-d** [comment]
> The client link on the sidebar no longer works because the client route changed

This will be changed in (#78) along with client list functionality

**Pull request merged**